tests for subsetByMonth

mhines-usgs opened the pull request.

mhines-usgs committed:
duh, wrong variable. copy pasta mmm.

mhines-usgs committed:
tests for subsetByMonth

thongsav-usgs commented on tests/testthat/test-utils-filter.R:
```diff
@@ -104,4 +104,23 @@ test_that('do negative or zero values get removed from the data frame?', {
+test_that('does it subset by the month I ask it to', {
+  library(jsonlite)
+  library(lubridate)
+  pts <- fromJSON(system.file('extdata','uvhydrograph','uvhydro-example.json', package = 'repgen'))
```
Instead of using the example file, I'd copy and paste the just enough of the data here and wittle it down to just a few points (eg: 1 point for each month). That way the test explicitly shows what it is testing without an external dependency.

EG:  https://github.com/USGS-R/repgen/blob/master/tests/testthat/test-utils-fetch.R#L6

mhines-usgs committed:
using small representative json instead of whole files

mhines-usgs committed:
conflict

mhines-usgs committed:
merge conflict

Merge branch 'master' of github.com:mhines-usgs/repgen

# Conflicts:
#	tests/testthat/test-utils-filter.R
